### PR TITLE
Introduce linter, taskfile and fix lint warnings

### LINKED ISSUE #45: Consider adding golangci-lint
[golangci-lint](https://golangci-lint.run/) is a Go linter aggregator that provides easy and quick access to a set of configurable Go linters that inspect the code and issue various warnings, ultimately helping to write better Go code.

A couple of examples:
```
internal/middleware/auth.go:43:8: ineffectual assignment to err (ineffassign)
                                _, err = bun.NewUpdate().Model(&session).Set("deleted_at = ?", time.Now()).Where("id = ?", session.ID).Exec(c.Request().Context())
```
(not checking the error)

```
internal/services/report_generation/generator/pdf/helper.go:50:2: S1021: should merge variable declaration with assignment on next line (gosimple)
        var ioReader io.Reader
```
(simplify the code)

```
internal/mail/template.go:61:9: S1025: should use String() instead of fmt.Sprintf (gosimple)
        return fmt.Sprintf("%s", out), nil
```
(parsing the format template is expensive, and since it does not contain anything other than `%s`, it can be replaced with String())

If you feel like this is a valuable addition to the project, feel free to assign this issue to me!

### PR DESCRIPTION
**Pull request by felixhromadko**
Aims to close #45

* changes the golang package from `example` to `github.com/dokedu/dokedu/backend` (please edit your run configuration)
* adds a taskfile for easy task running
* removes `restart: always` from the docker-compose db container

### CODE REVIEW

**Comment by tomhaerter** [comment]
Thank you @felixhromadko! I'll review it later today or tomorrow

File: backend/docker-compose.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ name: dokedu
 services:
   db:
     image: postgres:16-alpine
-    restart: always
```
Comment (tomhaerter) [inline backend/docker-compose.yml:7]: Out of curiosity, why did you remove the auto restart from the postgres container?

**Review by tomhaerter — APPROVED** [review]
Will be merged shortly. I checked it and it looks great, thanks also for testing everything thoroughly

File: backend/docker-compose.yml (hunk 1 above)
Comment (felixhromadko) [reply backend/docker-compose.yml:7]: The restart flag is a pretty important one for server or production deployments. If the host gets restarted, docker updated or the container crashes, it will automatically get restarted.

However, this also means that this container starts every time I turn my PC on, and blocks the default 5432 port, which is pretty annoying when I am working on other projects that also need port 5432

Comment (tomhaerter) [reply backend/docker-compose.yml:7]: Gotcha! Thanks for the explanation and removal. I used to stop the containers when switching between projects, but I also rarely restart my PC
